src/utils/stage2decorators.ts

```diff
@@ -0,0 +1,13 @@
+export type Stage2Decorator = {
+    kind: "field" | "method" | "class"
+    key: string
+    placement: "static" | "prototype" | "own"
+    descriptor: PropertyDescriptor
+    initializer?: () => any
+    finisher?: (klass) => void
+    extras?: Stage2Decorator[]
+}
+
+export function quacksLikeAStage2Decorator(args: IArguments): boolean {
+    return args.length === 1 && args[0] && (args[0].kind === "field" || args[0].kind === "method")
```

Comment:
I think that it would be more future-proof to also check if `args[0].kind === "class"`, just in case someone tries to use this helper with a class decorator.